modules/siddhi-core/src/main/java/org/wso2/siddhi/core/stream/output/sink/OutputMapper.java

```diff
@@ -40,13 +47,36 @@ public final void init(StreamDefinition streamDefinition,
                            String type,
                            OptionHolder mapOptionHolder,
                            String unmappedPayload,
-                           OptionHolder transportOptionHolder) {
+                           OptionHolder transportOptionHolder,
+                           ExecutionPlanContext executionPlanContext) {
         this.optionHolder = mapOptionHolder;
         this.type = type;
         if (unmappedPayload != null && !unmappedPayload.isEmpty()) {
             payloadTemplateBuilder = new TemplateBuilder(streamDefinition, unmappedPayload);
         }
+        this.elementId = executionPlanContext.getElementIdGenerator().createNewId();
         init(streamDefinition, mapOptionHolder, payloadTemplateBuilder);
+        executionPlanContext.getSnapshotService().addSnapshotable("outputmappers", this);
+    }
+
+    /**
+     * Updates the {@link Event#id}
+     *
+     * @param event event to be updated
+     */
+    public void updateEventId(Event event) {
+        event.setId(lastEventId.incrementAndGet());
+    }
+
+    /**
+     * Update the {@link Event#id}s
+     *
+     * @param events events to be updated
+     */
+    public void updateEventIds(Event[] events) {
```

Comment:
"updateEventsId" sounds a bit weird :) .. in the usual way we name things like this, it should be "updateEventIds". The idea is, the entity we are updating here is the "EventId" .. so the plural of that is "EventIds".